tests/test_trait_pipelines.py

```diff
@@ -1320,51 +1332,392 @@ def test_older_monocot_pipeline(
+    assert pd.DataFrame([computed_traits_6039_1["summary_stats"]]).shape == (1, 315)
+
+    assert isinstance(computed_traits_9535_1, dict)
+    assert computed_traits_9535_1["series"] == "9535_1"
+    assert computed_traits_9535_1["group"] == "9535"
+    assert pd.DataFrame([computed_traits_6039_1["traits"]]).shape == (1, 35)
+    assert pd.DataFrame([computed_traits_9535_1["summary_stats"]]).shape == (1, 315)
+
+    angle_traits = (
+        "lateral_angles_proximal",
+        "lateral_angles_distal",
+        "primary_angle_distal",
+        "primary_angle_proximal",
+    )
+
+    expected_dtypes = (int, float, np.integer, np.floating)
+
+    all_series_summaries = []
+
+    for series in all_multiple_dicot_series:
+
+        # Compute traits for the current series.
+        computed_traits = multiple_dicot_pipeline.compute_multiple_dicots_traits(series)
+
+        # Manually create dictionary storing traits for all frames.
+        result = {
+            "series": str(series.series_name),
+            "group": str(series.group),
+            "qc_fail": series.qc_fail,
+            "traits": {},
+            "summary_stats": {},
+        }
+
+        aggregated_traits = {}
+
+        for frame in range(len(series)):
+
+            frame_traits = {
+                "primary_pts": series.get_primary_points(frame),
+                "lateral_pts": series.get_lateral_points(frame),
+                "expected_plant_ct": series.expected_count,
+            }
+
+            frame_traits["primary_pts_no_nans"] = filter_roots_with_nans(
+                frame_traits["primary_pts"]
+            )
+            frame_traits["lateral_pts_no_nans"] = filter_roots_with_nans(
+                frame_traits["lateral_pts"]
+            )
+            frame_traits["filtered_pts_expected_plant_ct"] = (
+                filter_plants_with_unexpected_ct(
+                    frame_traits["primary_pts_no_nans"],
+                    frame_traits["lateral_pts_no_nans"],
+                    frame_traits["expected_plant_ct"],
+                )
+            )
+            frame_traits["primary_pts_expected_plant_ct"] = get_filtered_primary_pts(
+                frame_traits["filtered_pts_expected_plant_ct"]
+            )
+            frame_traits["lateral_pts_expected_plant_ct"] = get_filtered_lateral_pts(
+                frame_traits["filtered_pts_expected_plant_ct"]
+            )
+            frame_traits["plant_associations_dict"] = associate_lateral_to_primary(
+                frame_traits["primary_pts_expected_plant_ct"],
+                frame_traits["lateral_pts_expected_plant_ct"],
+            )
+
+            dicot_pipeline = DicotPipeline()
 
-    # Check that series dictionary
-    assert isinstance(arabidopsis_traits, dict)
-    assert arabidopsis_traits["series"] == "997_1"
-    assert arabidopsis_traits["group"] == "997"
+            # Extract the plant associations for this frame
+            associations = frame_traits["plant_associations_dict"]
```

Comment:
add a check here that associations are as expected. is this in the fixture? you know the number of associations, and the shape of points for each association. you could also check the primary points and lateral points for each plant.